cap/modules/auth/views.py

```diff
@@ -0,0 +1,156 @@
+from flask import url_for
+
+from flask import Blueprint, current_app, jsonify, request, redirect, session
+from flask_login import current_user
+from invenio_db import db
+from sqlalchemy.orm.attributes import flag_modified
+
+from .proxies import current_auth
+from .models import OAuth2Token
+
+from invenio_userprofiles.models import UserProfile
+from .config import OAUTH_SERVICES
+
+blueprint = Blueprint(
+    'cap_auth',
+    __name__,
+    url_prefix='/auth',
+)
+
+
+def ANAUTHORIZED_MSG(name):
+    return "You need to be logged in to connect your account with '{}'" \
+            .format(name)
+
+
+@blueprint.route('/connect/<name>')
+def connect(name):
+    if not current_user.is_authenticated:
+        return jsonify({"error": ANAUTHORIZED_MSG(name)}), 401
+
+    next_url = request.args.get('next')
+    session['next'] = next_url or None
+    ui_flag = request.args.get('ui')
+    session['ui'] = ui_flag or None
+
+    client = current_auth.create_client(name)
+    redirect_uri = url_for('cap_auth.authorize', name=name, _external=True)
+    return client.authorize_redirect(redirect_uri)
+
+
+@blueprint.route('/authorize/<name>')
+def authorize(name):
+    if not current_user.is_authenticated:
+        return jsonify({"error": ANAUTHORIZED_MSG(name)}), 401
+
+    ui_flag = session.pop('ui', None)
+
+    client = current_auth.create_client(name)
+    token = client.authorize_access_token()
+
+    configs = OAUTH_SERVICES.get(name.upper(), {})
+    extra_data_method = configs.get('extra_data_method')
+
+    # TOFIX Add error handlers for reject, auth errors, etc
+    extra_data = {}
+    if (extra_data_method):
+        extra_data = extra_data_method(client, token)
+
+    _token = OAuth2Token.get(name=name, user_id=current_user.id)
+    if not _token:
+        _token = OAuth2Token(name=name, user_id=current_user.id)
+    _token.token_type = token.get('token_type', 'bearer')
+    _token.access_token = token.get('access_token')
+    _token.refresh_token = token.get('refresh_token')
+    _token.expires_at = token.get('expires_at', 0)
+    _token.expires_in = token.get('expires_in', 0)
+    _token.extra_data = extra_data
+    db.session.add(_token)
```

Comment:
exactly what Oauth.update_token does
```suggestion
    _token = current_auth.update_token(name, token)
```